File: java/resistor-color/ResistorColor.java
import java.util.Arrays;

/**
 * Resistor Color
 * https://exercism.org/tracks/java/exercises/resistor-color
 */
class ResistorColor {

    private static final String[] ENCODING = {
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white"
    };

    int colorCode(String color) {
        for (int i = 0; i < ENCODING.length; i++) {
            if (color.equals(ENCODING[i])) {
                return i;
            }
        }

        return -1;
    }

    String[] colors() {
        return Arrays.copyOf(ENCODING, ENCODING.length);
    }
}

File: c/resistor-color/resistor_color.c
#include "resistor_color.h"

/**
 * Resistor Color
 * https://exercism.org/tracks/c/exercises/resistor-color
 */

uint16_t color_code(resistor_band_t band)
{
    return band;
}

resistor_band_t *colors()
{
    static resistor_band_t colors[] = {
        BLACK,
        BROWN,
        RED,
        ORANGE,
        YELLOW,
        GREEN,
        BLUE,
        VIOLET,
        GREY,
        WHITE
    };

    return colors;
}
